pandas/_libs/tslibs/period.pyx

```diff
@@ -1158,7 +1158,9 @@ cdef str period_format(int64_t value, int freq, object fmt=None):
         return "NaT"
 
     if isinstance(fmt, str):
-        fmt = fmt.encode("utf-8")
+        # Encode using current locale, in case fmt contains non-utf8 chars
+        # fmt = fmt.encode("utf-8")
```

Comment:
Looks unneeded?

Reply:
indeed. This was a reminder of the "old unwanted way". Removing it